### PR TITLE
refactor: Update performance recorder implementation

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: lib/commands/performance.js [hunk 1]
```diff
@@ -13,31 +16,179 @@ const PERF_RECORD_FEAT_NAME = 'perf_record';
...
+    this._zippedReportPath = zippedReportPath;
+    return this._zippedReportPath;
+  }
+
+  isRunning () {
+    return !!(this._process?.isRunning);
+  }
+
+  async _enforceTermination () {
+    if (this._process && this.isRunning()) {
```
Comment (imurchie) [inline lib/commands/performance.js]: `this._process` is not necessary here.

File: lib/commands/performance.js [hunk 2]
```diff
@@ -13,31 +16,179 @@ const PERF_RECORD_FEAT_NAME = 'perf_record';
...
+        await this._process.stop('SIGKILL');
+      } catch (ign) {}
+    }
+    this._process = null;
+    if (this._archivePromise) {
+      this._archivePromise
+        // eslint-disable-next-line promise/prefer-await-to-then
+        .then(async () => {
+          await fs.rimraf(this._zippedReportPath);
+          this._archivePromise = null;
```
Comment (imurchie) [inline lib/commands/performance.js]: Perhaps this should be in a `finally` block, to ensure it gets nulled?

File: lib/commands/performance.js [hunk 3]
```diff
@@ -13,31 +15,179 @@ const PERF_RECORD_FEAT_NAME = 'perf_record';
...
+        })
+        .catch(() => {});
+    } else {
+      if (await fs.exists(this._zippedReportPath)) {
```
Comment (imurchie) [inline lib/commands/performance.js]: `fs.rimraf` silently passes over non-existent paths, so there is no need to check before calling.

File: lib/commands/performance.js [hunk 4]
```diff
@@ -13,31 +15,179 @@ const PERF_RECORD_FEAT_NAME = 'perf_record';
...
+        await fs.rimraf(this._zippedReportPath);
+      }
+    }
+    if (await fs.exists(this._reportPath)) {
```
Comment (imurchie) [inline lib/commands/performance.js]: Ditto.

File: lib/commands/performance.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/commands/performance.js]: I've added it deliberately for better readability.

File: lib/commands/performance.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/commands/performance.js]: moved

File: lib/commands/performance.js (hunk 3 above)
Comment (mykola-mokhnach) [reply lib/commands/performance.js]: does it handle an empty string as non existent path?

Comment (imurchie) [reply lib/commands/performance.js]: As long as it's a string. It can't be `null` or `undefined`.

File: lib/commands/performance.js (hunk 1 above)
Comment (imurchie) [reply lib/commands/performance.js]: Not sure I understand how that's more readable. But ok.

**Pull request merged**